pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -1051,6 +1051,8 @@ int tracepoint__net__net_dev_queue(struct net_dev_queue_ctx* ctx) {
     sock_tup.pid = 0;
 
     if (!is_equal(&skb_tup, &sock_tup)) {
+        normalize_tuple(&skb_tup);
+        normalize_tuple(&sock_tup);
```

Comment:
because now `connection_protocol` is always indexed by a a normalized tuple.

let's say that `normalized(a, b) == normalized(b, a) == (a,b)`
when we're trying to fetch the protocol stack for a given connection `(a,b)` or `(b,a)` we'll use `(a,b)` as the map key; if the connection is not translated, we'll get back the protocol stack.

if the connection is NATed, the first lookup will fail and the result of `conn_tuple_to_socket_skb_conn_tuple` lookup will also be a normalized tuple, and used during a second lookup in the `connetion_protocol` map